CHANGELOG.md

```diff
@@ -1,5 +1,168 @@
 # Changelog
 
+## [v0.17.0](https://github.com/delta-io/delta-kernel-rs/tree/v0.17.0/) (2025-11-10)
+
+[Full Changelog](https://github.com/delta-io/delta-kernel-rs/compare/v0.16.0...v0.17.0)
+
+### 🏗️ Breaking changes
+1. (catalog-managed): New copy_atomic StorageHandler method ([#1400])
+2. Make expression and predicate evaluator constructors fallible ([#1452])
+3. (catalog-managed): add `log_tail` to `SnapshotBuilder` ([#1290])
+   - `log_tail` may optionally be passed to the `SnapshotBuilder`
+4. Arrow 57, MSRV 1.85+ ([#1424])
+5. Add ffi for idempotent write primitives ([#1191])
```

Comment:
Was this breaking or purely addative?

Reply:
added comment for each breaking change. This PR added Optional<> to the signature of an existing public FFI function.